shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Generator/Utxo.hs

```diff
@@ -210,61 +175,222 @@ genTxRetry
       -------------------------------------------------------------------------
       let wits = spendWits ++ wdrlWits ++ certWits ++ updateWits
           scripts = mkScriptWits spendScripts (certScripts ++ wdrlScripts)
-          mkTxWits' =
-            mkTxWits
-              ksIndexedPaymentKeys
-              ksIndexedStakingKeys
-              wits
-              scripts
-              . hashAnnotated
+          mkTxWits' = mkTxWits ksIndexedPaymentKeys ksIndexedStakingKeys wits scripts . hashAnnotated
       -------------------------------------------------------------------------
       -- SpendingBalance, Output Addresses (including some Pointer addresses)
       -- and a Outputs builder that distributes the given balance over addresses.
       -------------------------------------------------------------------------
-      let spendingBalance =
-            spendingBalanceUtxo
-              <> (fold (snd <$> wdrls)) Val.~~ deposits
-              <> refunds
+      let spendingBalance = spendingBalanceUtxo  <+> (inject $ ((sumVal (snd <$> wdrls)) <-> deposits) <+> refunds)
+          n = if (Map.size . unUTxO) utxo < 100 then 3 else 0
```

Comment:
should we maybe make this constants somewhere? add to the test parameter class? or just give them names so that they don't seem so much like magic numbers?

Reply:
I have added to the defaultConstants
  -- | When generating Tx, we want the UTxO size to fluctuate around this point. If
    --   it gets too small, we can't balance the fee, too large it gets too complicated.
    genTxStableUtxoSize:: Int,
    -- | If we need to grow the Utxo when generating a Tx, how much should it grow by.
    genTxUtxoIncrement:: Int